File: 1. Learn the Basics/1. Pattern/C++/17.cpp
/*

     A     
    ABA    
   ABCBA   
  ABCDCBA  
 ABCDEDCBA 

*/

#include <bits/stdc++.h>
using namespace std;

void alphaHill(int n) {
    // Write your code here
    for(int i=0;i<n;i++){
        for(int j=0;j<n-i-1;j++){
            cout << " ";
        }
        char letter = 'A';
        int breakpoint = (2*i+1)/2;
        for(int j=0;j<2*i+1;j++){
            cout << letter;
            if(j<breakpoint)
                letter++;
            else
                letter--;
        }
        for(int j=0;j<n-i-1;j++){
            cout << " ";
        }
        cout << endl;
    }
}
int main() {
	// your code goes here
	int n;
	cin >> n;
	alphaHill(n);
	return 0;
}

File: 1. Learn the Basics/1. Pattern/Python/17.py
#      A     
#     ABA    
#   ABCBA   
#   ABCDCBA  
#  ABCDEDCBA 

def alphaHill(n):
    for i in range(0,n):
        for j in range(0,n-i):
            print(" ",end="")
        letter = 65
        breakpoint = (2*i+1)/2
        for j in range (0,(2*i+1)):
            print(chr(letter),end="")
            if j < breakpoint-1:
                letter = letter+1
            else:
                letter = letter-1
        for j in range(0,n-i):
            print(" ",end="")
        print("\r")

        
n = int(input())
alphaHill(n)